File: ZSDK_DevDemos_Combined/DeveloperDemosCSharp/ConnectionPanel.cs
/********************************************** 
 * CONFIDENTIAL AND PROPRIETARY 
 *
 * The source code and other information contained herein is the confidential and the exclusive property of
 * ZIH Corp. and is subject to the terms and conditions in your end user license agreement.
 * This source code, and any other information contained herein, shall not be copied, reproduced, published, 
 * displayed or distributed, in whole or in part, in any medium, by any means, for any purpose except as
 * expressly permitted under such license agreement.
 * 
 * Copyright ZIH Corp. 2010
 *
 * ALL RIGHTS RESERVED 
 ***********************************************/ 

using System;
using System.Windows.Forms;
using System.Drawing;
using System.IO.Ports;

namespace ZebraDeveloperDemos {
    public partial class ConnectionPanel : UserControl {
        public ConnectionPanel() {
            InitializeComponent();
			connectionMethodCombo.SelectedIndex = 0;
            this.SerialBaudRate = 9600;
            this.SerialComPortName = "COM1";
            this.SerialDataBits = 8;
            this.SerialHandshake = System.IO.Ports.Handshake.None;
            this.SerialParity = System.IO.Ports.Parity.None;
            this.SerialStopBits = System.IO.Ports.StopBits.One;
        }

        public void SetStatus(String message, Color color) {
            statusLabel.Text = message;
            statusLabel.BackColor = color;
        }
        
        public String ConnectionType {
            get {
                return connectionMethodCombo.Text;
            }
            set {
            }
        }

        public String IPAddress {
            get {
                return ipDnsAddress.Text;
            }
            set {
                ipDnsAddress.Text = value;
            }
        }

        public int PortNum {
            get {
                int retVal = 6101;
                if (portNumber.Text.Equals("") == false) {
                    retVal = Convert.ToInt32(portNumber.Text);
                }
                return retVal;
            }
            set {
                portNumber.Text = Convert.ToString(value);
            }
        }

        public String BluetoothMacAddress {
            get {
                return macAddress.Text;
            }
            set {
                macAddress.Text = value;
            }
        }
        public String UsbPortName {
            get {
                String retVal = "LPT1";
                if (UsbPortNameCombo.Text.Equals("") == false) {
                    retVal = UsbPortNameCombo.Text;
                }
                return retVal;
            }
        }
        public String SerialComPortName {
            get {
                String retVal = "COM1";
                if (portNamesCombo.Text.Equals("") == false) {
                    retVal = portNamesCombo.Text;
                }
                return retVal;
            }
            set {
                portNamesCombo.Text = value;
            }
        }

        public int SerialBaudRate {
            get {
                int retVal = 9600;
                if (baudRateCombo.Text.Equals("") == false) {
                    retVal = Convert.ToInt32(baudRateCombo.Text);
                }
                return retVal;
            }
            set {
                baudRateCombo.Text = Convert.ToString(value);
            }
        }

        public int SerialDataBits {
            get {
                int retVal = 8;
                if (dataBitsCombo.Text.Equals("") == false) {
                    retVal = Convert.ToInt32(dataBitsCombo.Text);
                }
                return retVal;
            }
            set {
                dataBitsCombo.Text = Convert.ToString(value);
            }
        }

        public Parity SerialParity {
            get {
                Parity retVal = Parity.None;
                if (parityCombo.Text.Equals("") == false) {
                    retVal = (Parity)Enum.Parse(typeof(Parity), parityCombo.Text, true);
                }
                return retVal;
            }
            set {
                parityCombo.Text = value.ToString();
            }
        }

        public StopBits SerialStopBits {
            get {
                return convertStopBitsTextToEnum(stopBitsCombo.Text);
            }
            set {
                stopBitsCombo.Text = convertStopBitsEnumToText(value);
            }
        }

        private String convertStopBitsEnumToText(StopBits value) {
            String retVal = "1";
            switch (value) {
                case StopBits.OnePointFive:
                    retVal = "1.5";
                    break;
                case StopBits.Two:
                    retVal = "2";
                    break;
                case StopBits.One:
                default:
                    retVal = "1";
                    break;
            }

            return retVal;
        }

        private StopBits convertStopBitsTextToEnum(String stopBitsText) {
            StopBits retVal = StopBits.One;
            if (stopBitsText.Equals("1.5") == true) {
                retVal = StopBits.OnePointFive;
            } else if (stopBitsText.Equals("2") == true) {
                retVal = StopBits.Two;
            }
            return retVal;
        }

        public Handshake SerialHandshake {
            get {
                Handshake retVal = Handshake.None;
                if (handshakeCombo.SelectedIndex > -1) {
                    retVal = (Handshake)handshakeCombo.SelectedIndex;
                }
                return retVal;
            }
            set {
                handshakeCombo.Text = value.ToString();
            }
        }
        private void connectionMethod_SelectedValueChanged(object sender, EventArgs e) {
            String selectedConnectionMethod = ((ComboBox)sender).Text;
            
            ipDnsPanel.Visible = false;
            bluetoothPanel.Visible = false;
            serialPanel.Visible = false;
            usbPanel.Visible = false;

            switch (selectedConnectionMethod) {
                case "IP/DNS":
                    ipDnsPanel.Visible = true;
                    break;
                case "Bluetooth(R)":
                    bluetoothPanel.Visible = true;
                    break;
                case "Serial":
                    serialPanel.Visible = true;
                    break;
                case "USB":
                    usbPanel.Visible = true;
                    break;
                default:
                    break;
            }
        }

        private void portNumber_KeyPress(object sender, KeyPressEventArgs e) {
            //numeric values only
            if((e.KeyChar < '0' || e.KeyChar > '9') && e.KeyChar != (char)Keys.Back)
            {
                e.Handled = true;
            }
            base.OnKeyPress(e);
        }
    }
}

File: ZSDK_DevDemos_Combined/DeveloperDemosVisualBasic/ConnectionPanel.vb
' ********************************************** 
' * CONFIDENTIAL AND PROPRIETARY 
' *
' * The source code and other information contained herein is the confidential and the exclusive property of
' * ZIH Corp. and is subject to the terms and conditions in your end user license agreement.
' * This source code, and any other information contained herein, shall not be copied, reproduced, published, 
' * displayed or distributed, in whole or in part, in any medium, by any means, for any purpose except as
' * expressly permitted under such license agreement.
' * 
' * Copyright ZIH Corp. 2010
' *
' * ALL RIGHTS RESERVED 
' *********************************************** 

Imports System
Imports System.Windows.Forms
Imports System.Drawing
Imports System.IO.Ports

Namespace ZebraDeveloperDemos
    <System.ComponentModel.DesignerCategory("Code")> _
    Partial Public Class ConnectionPanel
        Inherits UserControl
        Public Sub New()
            InitializeComponent()
            connectionMethodCombo.SelectedIndex = 0
            Me.SerialBaudRate = 9600
            Me.SerialComPortName = "COM1"
            Me.SerialDataBits = 8
            Me.SerialHandshake = System.IO.Ports.Handshake.None
            Me.SerialParity = System.IO.Ports.Parity.None
            Me.SerialStopBits = System.IO.Ports.StopBits.One
        End Sub

        Public Sub SetStatus(ByVal message As [String], ByVal color As Color)
            statusLabel.Text = message
            statusLabel.BackColor = color
        End Sub

        Public Property ConnectionType() As [String]
            Get
                Return connectionMethodCombo.Text
            End Get
            Set(ByVal value As [String])
            End Set
        End Property

        Public Property IPAddress() As [String]
            Get
                Return ipDnsAddress.Text
            End Get
            Set(ByVal value As [String])
                ipDnsAddress.Text = Value
            End Set
        End Property

        Public Property PortNum() As Integer
            Get
                Dim retVal As Integer = 6101
                If portNumber.Text.Equals("") = False Then
                    retVal = Convert.ToInt32(portNumber.Text)
                End If
                Return retVal
            End Get
            Set(ByVal value As Integer)
                portNumber.Text = Convert.ToString(value)
            End Set
        End Property

        Public Property BluetoothMacAddress() As [String]
            Get
                Return macAddress.Text
            End Get
            Set(ByVal value As [String])
                macAddress.Text = value
            End Set
        End Property
        Public ReadOnly Property UsbPortName() As [String]
            Get
                Dim retVal As [String] = "LPT1"
                If UsbPortNameCombo.Text.Equals("") = False Then
                    retVal = UsbPortNameCombo.Text
                End If
                Return retVal
            End Get
        End Property
        Public Property SerialComPortName() As [String]
            Get
                Dim retVal As [String] = "COM1"
                If portNamesCombo.Text.Equals("") = False Then
                    retVal = portNamesCombo.Text
                End If
                Return retVal
            End Get
            Set(ByVal value As [String])
                portNamesCombo.Text = value
            End Set
        End Property

        Public Property SerialBaudRate() As Integer
            Get
                Dim retVal As Integer = 9600
                If baudRateCombo.Text.Equals("") = False Then
                    retVal = Convert.ToInt32(baudRateCombo.Text)
                End If
                Return retVal
            End Get
            Set(ByVal value As Integer)
                baudRateCombo.Text = Convert.ToString(value)
            End Set
        End Property

        Public Property SerialDataBits() As Integer
            Get
                Dim retVal As Integer = 8
                If dataBitsCombo.Text.Equals("") = False Then
                    retVal = Convert.ToInt32(dataBitsCombo.Text)
                End If
                Return retVal
            End Get
            Set(ByVal value As Integer)
                dataBitsCombo.Text = Convert.ToString(value)
            End Set
        End Property

        Public Property SerialParity() As Parity
            Get
                Dim retVal As Parity = Parity.None
                If parityCombo.Text.Equals("") = False Then
                    retVal = CType([Enum].Parse(GetType(Parity), parityCombo.Text, True), Parity)
                End If
                Return retVal
            End Get
            Set(ByVal value As Parity)
                parityCombo.Text = value.ToString()
            End Set
        End Property

        Public Property SerialStopBits() As StopBits
            Get
                Return convertStopBitsTextToEnum(stopBitsCombo.Text)
            End Get
            Set(ByVal value As StopBits)
                stopBitsCombo.Text = convertStopBitsEnumToText(value)
            End Set
        End Property

        Private Function convertStopBitsEnumToText(ByVal value As StopBits) As [String]
            Dim retVal As [String] = "1"
            Select Case value
                Case StopBits.OnePointFive
                    retVal = "1.5"
                    Exit Select
                Case StopBits.Two
                    retVal = "2"
                    Exit Select
                Case StopBits.One
                Case Else
                    retVal = "1"
                    Exit Select
            End Select

            Return retVal
        End Function

        Private Function convertStopBitsTextToEnum(ByVal stopBitsText As [String]) As StopBits
            Dim retVal As StopBits = StopBits.One
            If stopBitsText.Equals("1.5") = True Then
                retVal = StopBits.OnePointFive
            ElseIf stopBitsText.Equals("2") = True Then
                retVal = StopBits.Two
            End If
            Return retVal
        End Function

        Public Property SerialHandshake() As Handshake
            Get
                Dim retVal As Handshake = Handshake.None
                If handshakeCombo.SelectedIndex > -1 Then
                    retVal = CType(handshakeCombo.SelectedIndex, Handshake)
                End If
                Return retVal
            End Get
            Set(ByVal value As Handshake)
                handshakeCombo.Text = value.ToString()
            End Set
        End Property
        Private Sub connectionMethod_SelectedValueChanged(ByVal sender As Object, ByVal e As EventArgs)
            Dim selectedConnectionMethod As [String] = DirectCast(sender, ComboBox).Text

            ipDnsPanel.Visible = False
            bluetoothPanel.Visible = False
            serialPanel.Visible = False
            usbPanel.Visible = False

            Select Case selectedConnectionMethod
                Case "IP/DNS"
                    ipDnsPanel.Visible = True
                    Exit Select
                Case "Bluetooth(R)"
                    bluetoothPanel.Visible = True
                    Exit Select
                Case "Serial"
                    serialPanel.Visible = True
                    Exit Select
                Case "USB"
                    usbPanel.Visible = True
                    Exit Select
                Case Else
                    Exit Select
            End Select
        End Sub

        Private Sub portNumber_KeyPress(ByVal sender As Object, ByVal e As KeyPressEventArgs)
            'numeric values only
            If (e.KeyChar < "0"c OrElse e.KeyChar > "9"c) AndAlso e.KeyChar <> ChrW(Keys.Back) Then
                e.Handled = True
            End If
            MyBase.OnKeyPress(e)
        End Sub
    End Class
End Namespace
